src/datajudge/requirements.py

```diff
@@ -606,18 +608,42 @@ def add_varchar_regex_constraint(
         ``n_counterexamples`` defines how many counterexamples are displayed in an
         assertion text. If all counterexamples are meant to be shown, provide ``-1`` as
         an argument.
+
+        ``computation_in_db`` defines whether the regex matching takes place in database or
+        in memory. Typically the former is both faster and substantially more memory-saving.
+        Yet, it is only supported for Postgres, Sqllite and Snowflake. Note that for this
+        feature is only for Snowflake when using sqlalchemy-snowflake >= 1.4.0.
         """
         ref = DataReference(self.data_source, [column], condition)
-        self._constraints.append(
-            varchar_constraints.VarCharRegex(
-                ref,
-                regex,
-                allow_none=allow_none,
-                relative_tolerance=relative_tolerance,
-                aggregated=aggregated,
-                n_counterexamples=n_counterexamples,
+        if computation_in_db:
+            if allow_none:
+                warnings.warn(
+                    """
+                    Parameter 'allow_none' is not allowed when using 'computation_in_db'.
```

Comment:
From my perspective it would probably make sense to deprecate this parameter both when using `computation_in_db` and when not using it.